tests/terminator_tests/improvement_tests/test_preprocessing.py

```diff
@@ -0,0 +1,236 @@
+    trials_after = p.apply(trials_before, direction)
+
+    n_trials_expected = int(len(values) * top_trials_ratio)
+    assert len(trials_after) == n_trials_expected
+    assert _get_trial_values(trials_after) == values_in_order[:n_trials_expected]
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_to_minimize(direction: StudyDirection) -> None:
+    values = [1, 0, 0, 2, 1]
+    trials_before = [create_trial(value=v) for v in values]
+
+    expected_values = values[:]
+    if direction == StudyDirection.MAXIMIZE:
+        expected_values = [-1 * v for v in expected_values]
+
+    p = _preprocessing.ToMinimize()
+    trials_after = p.apply(trials_before, direction)
+    assert _get_trial_values(trials_after) == expected_values
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_one_to_hot(direction: StudyDirection) -> None:
+    # Scenario: categorical parameters are mapped to the float distribution.
+    trials_before = [
+        create_trial(
+            params={"categorical": 0},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+        create_trial(
+            params={"categorical": 1},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.OneToHot()
+    trials_after = p.apply(trials_before, direction)
+
+    for t in trials_after:
+        # Distributions are expected to be mapped to `FloatDistribution(0, 1)` for each choice.
+        assert len(t.distributions) == 2
+        for d in t.distributions.values():
+            assert d == FloatDistribution(0, 1)
+
+    assert trials_after[0].params == {"i0_categorical": 1.0, "i1_categorical": 0.0}
+    assert trials_after[1].params == {"i0_categorical": 0.0, "i1_categorical": 1.0}
+
+    # Scenario: int and float parameters are identical even after the preprocessing.
+    trials_before = [
+        create_trial(
+            params={
+                "float": 1,
+                "int": 3,
+            },
+            distributions={
+                "float": FloatDistribution(1, 10),
+                "int": IntDistribution(1, 10),
+            },
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.OneToHot()
+    trials_after = p.apply(trials_before, direction)
+
+    for name_before, param_before in trials_before[0].params.items():
+        distribution_before = trials_before[0].distributions[name_before]
+
+        # Note that the param/distribution names are modified with "i0_" prefix to be consistent
+        # with categorical parameters.
+        name_after = f"i0_{name_before}"
+        param_after = trials_after[0].params[name_after]
+        distribution_after = trials_after[0].distributions[name_after]
+
+        assert distribution_before == distribution_after
+        assert param_before == param_after
```

Comment:
This is not a strong opinion, but these test can be written similarly to the previous scenario.

```suggestion
    assert len(trials_after) == 1
    # Note that the param/distribution names are modified with "i0_" prefix to be consistent
    # with categorical parameters.
    assert trials_after[0].params == {
        "i0_float": 1,
        "i0_int": 3,
    }
    assert trials_after[0].distributions == {
        "i0_float": FloatDistribution(1, 10),
        "i0_int": IntDistribution(1, 10),
    }
```